### PR TITLE
V1200 read length parameter & guppy update

### PR DESCRIPTION
**Pull request by replikation**
@DataSpott should be done or?

### CODE REVIEW

File: poreCov.nf [hunk 1]
```diff
@@ -113,6 +113,8 @@ if (params.fasta == true) { exit 5, "Please provide a fasta file via [--fasta]"
 if (params.fastq == true) { exit 5, "Please provide a fastq files (one per sample) via [--fastq]" }
 if (params.fastq_pass == true) { exit 5, "Please provide a fastq_pass dir via [--fastq_pass]" }
 if (params.fast5 == true) { exit 5, "Please provide a fast5 dir via [--fast5]" }
+if (!params.minLength == false && !params.minLength.toString().matches("[0-9]+")) { exit 5, "Please provide an integer number (e.g. 300) as minimal read length via [--minLength]" }
```
Comment (replikation) [inline poreCov.nf:116]: * please change to `params.minLength`

File: workflows/process/guppy.nf [hunk 2]
```diff
@@ -73,7 +73,7 @@ process guppy_gpu {
 process guppy_cpu {
         label 'guppy_cpu'
         if (!params.localguppy && workflow.profile.contains('docker') || workflow.profile.contains('singularity') ) {
-            container = 'nanozoo/guppy_cpu:4.2.2-1--416f83d'
+            container = 'nanozoo/guppy_cpu:4.5.4-1--8a4421c'
```
Comment (replikation) [inline workflows/process/guppy.nf:76]: * replace with 5.0.7

File: poreCov.nf (hunk 1 above)
Comment (DataSpott) [reply poreCov.nf]: Changed
Exit-code also prevents user from using the [--min/maxLength]-flag without handing over a value

File: workflows/process/guppy.nf (hunk 2 above)
Comment (DataSpott) [reply workflows/process/guppy.nf]: added new container

**Comment by hoelzer** [comment]
@replikation worked on Singularity and my work station even w/ CUDA v10.1

I only observed slight differences in the consensus sequences. Mostly w/ the PR version some more reads are assigned to SC2 but sometimes the quality of the final consensus is a bit lower, for example:

v0.8.3
image

current PR
image

Still fine I would say and the shown case here is anyway edgy bc/ the N content was already close to 5% but just something I wanted to share. Do you generally see an improved performance w/ new Guppy?

File: poreCov.nf [hunk 3]
```diff
@@ -416,8 +417,8 @@ ${c_yellow}Parameters - nCov genome reconstruction${c_reset}
     --primerV       artic-ncov2019 primer_schemes [default: ${params.primerV}]
                         Supported: V1, V2, V3, V1200
     --rapid         use rapid-barcoding-kit [default: ${params.rapid}]
-    --minLength     min length filter raw reads [default: ${params.minLength}]
-    --maxLength     max length filter raw reads [default: ${params.maxLength}]
+    --minLength     min length filter raw reads [default: 350; 500 for V1200 primer scheme]
```
Comment (replikation) [inline poreCov.nf:420]: @DataSpott `[default: 350; 500 for V1200 primer scheme]` to `[default: 350; 500 for V3; V1200 primer scheme]`

Comment (DataSpott) [reply poreCov.nf]: applied a change to make it better understandable

**Comment by DataSpott** [comment]
Haben jetzt die PR-Version gegen die aktuelle Version getestet und einen massiven Read-drop gehabt:
|Status|Reads|Bases|N50|Median Read Length|Median PHRED score|
|-|-|-|-|-|-|
|All Reads actual Version|4.64e+5|5.12e+8|1230|1210|12.770|
|Pass Reads actual Version|4.22e+5|4.68e+8|1230|1210|13.088|
|All Reads PR-Version|2.46e+5|2.79e+8|1220|1200|14.444|
|Pass Reads PR-Version|2.44e+5|2.77e+8|1220|1210|14.479|

Es fehlen bei der PR-Version etwa 2.2e+5 Reads. Haben eine Vermutung (veränderter QScore-Threshold von Guppy) woran es liegt und gehen der jetzt nach.

**Comment by DataSpott** [comment]
Issue with the missing bases solved: was an error during the data transfer, so that not all reads reached the workflow.

Now we have a comparison of where we analyzed one of our batches with the actual version and the pr-version.
Result is that most samples are the same, except for one (under the ones that passed the QC-score). In this 3 more spike-protein mutations were detected and one from a mutation changed amino acid results in another ac than in the actual version (highlighted by bold-writing):

|sample|lineage|nextclade_mutations|
|-|-|-|
|6076a81505996601b7a4884e_pr_version|B.1.1.7|N:R203K,**N:G204P**,N:S235F,ORF1a:T1001I,ORF1a:P1497L,ORF1a:A1708D,ORF1a:I2230T,ORF1b:P314L,ORF1b:K1383R,ORF1b:P1727L,S:N501Y,S:A570D,S:D614G,**S:P681H**,**S:T716I**,**S:S982A**,S:D1118H|
|6076a81505996601b7a4884e_v0.8.3|B.1.1.7|N:R203K,**N:G204R**,N:S235F,ORF1a:T1001I,ORF1a:P1497L,ORF1a:A1708D,ORF1a:I2230T,ORF1b:P314L,ORF1b:K1383R,ORF1b:P1727L,S:N501Y,S:A570D,S:D614G,S:D1118H|

Will analyze some more batches to test if this was some kind of edge case or due to the higher accuracy.

**Comment by hoelzer** [comment]
@DataSpott thx! The additional three amino acid changes make sense in the context of B.1.1.7 (you are likely already aware of that)
https://outbreak.info/compare-lineages?gene=S&threshold=75

Maybe in that specific case, they are located in a low-coverage region and now w/ the new model some more reads are called and then its possible to also detect P681H, etc... But anyway, even if the region was covered with many reads in 0.8.3 already it's still interesting that w/ the new model these B.1.1.7-characteristic changes can be called.

**Comment by DataSpott** [comment]
Ran a comparison between Medaka and Nanopolish using the new Guppy-version:
Test was performed using a batch with 24 samples from our routine -> 13 samples passed the QC-test, 10 failed it and the last sample was the negative control.

The following table includes all samples that differed between the two models:
|_id|medaka|nanopolish|QC valid|
|-|-|-|-|
|60914e148fc9b8496fa64221|189 N´s|190 N´s|true|
|60914e148fc9b8496fa64223|S:N354K;37 identity mismatches|S:N354X;36 identity mismatches|true|
|60914e148fc9b8496fa6422b|20205 N´s|20204 N´s|false|
|60914e148fc9b8496fa6422e|S:H69-, S:V70-, S:Y144-; 14783 N´s|missing deletions;14794 N´s|false|
|60914e148fc9b8496fa6422f|4021 N´s|4023 N´s|false|
|609270253d108d14e2765716|E:L21X;189 N´s|E:L21F;190 N´s|true|
609270253d108d14e2765716_2|E:L21X;190 N´s|E:L21F;232 N´s|true|

**Comment by hoelzer** [comment]
Interesting, so basically I would say the `medaka` looks better in terms of variant calls. E.g. `nanopolish` completely misses the important B.1.1.7-characteristic deletions in 60914e148fc9b8496fa6422e. 

But I'm not sure about the E:L21X vs L21F in the last two sequences. Normally, not much is going on in the E gene anyway (https://outbreak.info/compare-lineages?pango=B.1.1.7&pango=B.1.351&pango=B.1.617.2&pango=P.1&pango=B.1.427&pango=B.1.429&pango=B.1.526&pango=B.1.526.1&pango=B.1.526.2&pango=B.1.617&pango=B.1.617.1&pango=B.1.617.3&pango=P.2&pango=A.27&gene=E&threshold=75)

Do you also have the results for the older Guppy version and e.g. `medaka`? What was called here for this site in the E gene?

**Comment by replikation** [comment]
@hoelzer also as EL21X
